src/scrapers/leumi.ts

```diff
@@ -51,206 +46,105 @@ function createLoginFields(credentials: ScraperCredentials) {
 
-  const balanceValue = await page.$eval('#lblBalancesVal', (element) => element.textContent);
-  const balance = balanceValue ? parseFloat(balanceValue.replace(/[^\d.]/g, '')) : undefined;
+  // we must blur the from control otherwise the search will use the previous value
+  await page.focus("button[aria-label='סנן']");
 
-  await Promise.race([
-    waitUntilElementFound(page, 'table#WorkSpaceBox table#ctlActivityTable', false),
-    waitUntilElementFound(page, '.errInfo', false),
-  ]);
+  await clickButton(page, "button[aria-label='סנן']");
+  const finalResponse = await page.waitForResponse((response) => {
+    return response.url() === FILTERED_TRANSACTIONS_URL &&
+        response.request().method() === 'POST';
+  });
 
-  if (await isNoTransactionInDateRangeError(page)) {
-    return {
-      accountNumber,
-      balance,
-      txns: [],
-    };
-  }
+  const responseJson: any = await finalResponse.json();
 
-  const hasExpandAllButton = await elementPresentOnPage(page, 'a#lnkCtlExpandAllInPage');
+  const accountNumber = accountId.replace('/', '_');
+  const response = JSON.parse(responseJson.jsonResp);
 
-  if (hasExpandAllButton) {
-    await clickButton(page, 'a#lnkCtlExpandAllInPage');
-  }
+  const pendingTransactions = response.TodayTransactionsItems;
+  const transactions = response.HistoryTransactionsItems;
+  const balance = response.BalanceDisplay ? parseFloat(response.BalanceDisplay) : undefined;
 
-  const pendingTxns = await extractPendingTransactionsFromPage(page);
-  const completedTxns = await extractCompletedTransactionsFromPage(page);
+  const pendingTxns = extractTransactionsFromPage(pendingTransactions, TransactionStatuses.Pending);
+  const completedTxns = extractTransactionsFromPage(transactions, TransactionStatuses.Completed);
   const txns = [
     ...pendingTxns,
     ...completedTxns,
   ];
 
   return {
     accountNumber,
-    txns: convertTransactions(txns),
+    balance,
+    txns,
   };
 }
 
-async function fetchTransactions(page: Page, startDate: Moment) {
-  const res: TransactionsAccount[] = [];
-  // Loop through all available accounts and collect transactions from all
-  const accounts = await dropdownElements(page, 'select#ddlAccounts_m_ddl');
-  for (const account of accounts) {
-    // Skip "All accounts" option
-    if (account.value !== '-1') {
-      res.push(await fetchTransactionsForAccount(page, startDate, account.value));
-    }
+function hangProcess(timeout: number) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, timeout);
+  });
+}
+async function fetchTransactions(page: Page, startDate: Moment): Promise<TransactionsAccount[]> {
+  // TODO should adjust logic to support multiple accounts (I don't have such an account)
+
+  // DEVELOPER NOTICE the account number received from the server is being altered at
+  // runtime for some accounts after 1-2 seconds so we need to hang the process for a short while.
+  await hangProcess(4000);
```

Comment:
I don't like this - what's the value of the field before it gets updated?
we can easily do a validation that the account number is in the right format, and wait until that's the case.
we should also use the `waitUntil` helper function for this sleep.

Reply:
we cannot because this is a bug on their side that is being fixed and it can vary between users. for example I'm getting xxx-xxxxxx/46 which is then changed to xxx-xxxxxx/26 due to hard-coded logic that was added, I believe as a workaround to the wrong value returned from the server. @ifeins gets `/33`, there are multiple types and we cannot predict when they run their workaround.

I believe in the future we will be able to remove it, but at the moment, as much as I hate adding timeouts, this time I believe it is the best solution we have.

Reply:
@esakal got it - can we then just switch to using `waitUntil`?